sdv/constraints/utils.py

```diff
@@ -166,3 +166,30 @@ def revert_nans_columns(table_data, nan_column_name):
             table_data.loc[table_data[nan_column_name] == combination, column_names] = np.nan
 
     return table_data.drop(columns=nan_column_name)
+
+
+def get_datetime_diff(high, low, dtype='O'):
+    """Calculate the difference between two datetime columns.
+
+    When casting datetimes to float using ``astype``, NaT values are not automatically
+    converted to NaN values. This method calculates the difference between the high
+    and low column values, preserving missing values as NaNs.
+
+    Args:
+        high (numpy.ndarray):
+            The high column values.
+        low (numpy.ndarray):
+            The low column values.
+
+    Returns:
+        numpy.ndarray:
+            The difference between the high and low column values.
+    """
+    if dtype == 'O':
+        low = cast_to_datetime64(low)
+        high = cast_to_datetime64(high)
```

Comment:
minor: add line below

Reply:
Yes, done in [3a3cc68](https://github.com/sdv-dev/SDV/pull/1404/commits/3a3cc68a54677e2cca58f8417712fe9862497998).